### ISSUE TITLE
Is there a way to hook in remote queries?

### ISSUE THREAD

**Issue by aguynamedben** [plotly/react-pivottable#17]
Hi, I'm considering using this library, thanks for making it open source.

It seems like this library expects all the data to be loaded up-front and then thrown into the pivot table for pivoting. Because `data` [can be a callback](https://react-pivottable.js.org/), it seems very easy to make remote calls to load data before users do the pivoting.

I'm working on a analytics UI with higher volume data that wont fit in the DOM. The data is stored in a star schema in a MPP database. Is there a way this library can work as a front-end query builder? Up front the product would provide the available facts/dimensions that can be selected, then the user would select fact(s) and dimension(s) with drag and drop, and hit "run". At I'd get called with the facts/dimensions the user has selected and could ping off to my remove server for the results.

I guess maybe what I'm looking for is more of an OLAP query builder vs. an in-memory pivot table. I just wanted to see if you had any thoughts on that, or had seem people with similar requests. I know this maybe be beyond the scope of the roadmap for this project, but the UIs for the solutions seem very similar.

Thanks again for making this project public, it's really slick.

**Comment by nicolaskruchten** [comment]
Glad you like the UI! Unfortunately, as you've figured out, it expects all the data to be available up front and then takes over, there's no way for this component to have a conversation with a server or serve as a query-builder.

That said, the data doesn't all get dumped into the DOM, so that shouldn't be a restriction. So long as it can fit in RAM it should be OK. I've gone up to around 100k rows, which is big by some standards and small by others... How big is the dataset you're looking at?

**Comment by aguynamedben** [comment]
@nicolaskruchten Thanks for the perspective. It's a new project so I'm still getting a grasp on it, but it's at least a few million rows. There are other complicated things going on server-side so it's probably not plug-and-play. I went looking for a "build a BI" React library.

The closest I found was this: https://github.com/jjmontesl/cubesviewer which provides a decent query UI on top of this pretty decent looking Python cube: https://github.com/DataBrewery/cubes. I'm looking at using [Apache Kylin](http://kylin.apache.org/) to build OLAP cubes, but then got to thinking about the product side of building the UI to query the cube and that's where I don't see any fancy React projects.

Oh well, guess I might have to build it or find another way! Thanks again for the perspective. The UI you have is great, I'll let you know if I build anything influenced by it.
